include/ppx/application.h

```diff
@@ -409,6 +412,17 @@ class Application
 
     bool IsXrEnabled() const { return mSettings.xr.enable; }
 
+    // Starts a new metric run and returns it.
+    // Only one run is ever active at the same time.
+    // Default metrics are automatically added to the run: framerate, cpu_frame_time and frame_count.
+    // Additional ones may be added by the caller.
```

Comment:
> I am curious what this is supposed to look like in terms of updating the metrics, would each application have its own method similar to UpdateMetrics?

Yes, exactly. Each application can add whatever metrics they want to a run. Each run has default common metrics (framerate, frame time and frame count for now), but any other metrics is for the application to define. For example, let's say a foveation application wants to track image quality, it would 1) start a run, 2) add a custom "quality" metric to that run, 3) compute the image quality each frame (or whatever frequency needed) and 4) record the computed quality in the metric.

> The Start/Stop functions are not called by the parent class, meaning it's up to each application to call those. Meaning they are in control of the run lifecycle.

Yes, the applications control the runs. it is not really the lifecycle per-se. Runs as not deleted when they are stopped. This start/stop cycle cannot be handled by the parent class. An application might want to create several discrete runs, with different parameters. Also, applications will know when they load assets or do heavy work that should not be part of the run.

> So I see this as a default set of metrics you have for free, but then each application needs to manually call the manager to manager.addMetric, and handles them itself.

Yes, that was the exact original purpose. The application gets to opt-in for metrics and can then start runs, one at a time, because having several runs does not make sense anyway.

The pseudo-code is actually misleading. It is more like the following. The application does not interact with the manager, only with the runs it starts. Metrics are added to those runs, as not all runs would necessarily need the same metrics.

```
class MyApp : Application {
    Gauge customMetric;
    
    Setup() {
       settings.enableMetrics = true;
    }
    
    OnClick() {
       if(run_active) {
          StopRun();
       } else {
          Run run = StartRun();
          customMetric = run.AddMetricGauge();
       }
    }
    
    MainLoop() {
       customMetric .logValue(x);
    }
```

Again, as people use the metrics through the base application class, we can tweak the API. @Slumpwuffle already has some ideas, and since he's the earliest user, I am all in to let him shape the API.